src/Host-SDK/Temporal.Worker.Activities/ActivityApi.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Threading.Tasks;
+
+using Temporal.Common.DataModel;
+using Temporal.Serialization;
+
+namespace Temporal.Worker.Activities
+{
+    public class ActivityAPi
+    {
+    }
+
+    // =========== Basic activity abstractions ===========
+
+    public interface IBasicActivity
+    {
+        string ActivityTypeName { get; }
+
+        Task<PayloadsCollection> RunAsync(PayloadsCollection input, WorkflowActivityContext workflowCtx);        
```

Comment:
This is mentioned once elsewhere, but is it standard to postfix these names with `Async` in .NET? The type signature makes that clear, and it seems just `Run` would work fine here

Reply:
Unrelated, but important, I believe this interface should be generic over the input type. 

In your later implementation example:
```c#
public class SpeakAGreetingActivity : IBasicActivity
        {
            /// <summary>Optionally customize the name under which this activity is registered:</summary>
            public string ActivityTypeName { get { return "SpeakAGreeting1"; } }

            public Task<PayloadsCollection> RunAsync(PayloadsCollection input, WorkflowActivityContext activityCtx)
            {
                string greetingText = activityCtx.GetSerializer(input).Deserialize<string>(input) ?? "<null>";
                Console.WriteLine($"[{ActivityTypeName}] {greetingText}");
                return Task.FromResult<PayloadsCollection>(null);
            }
        }
```

You deserialize inside the activity. Users shouldn't have to do this, since the first line of basically every activity is going to be exactly that. This should be done for the user, and then the actual type they care about should be passed in to their activity function.

Reply:
So we'd end up with something like `public interface IActivity<T> where T: IDataValue` (assuming we do use `IDataValue`)

Reply:
Re Async - see pointers in the equivalent thread.

Reply:
Re generic iface: 
The idea is the same layering as with workflows: the lowest level of abstraction provides access to the raw PayloadsCollection. A higher level of abstraction is generic and required an IDavaValue parameter. We can add an IActivity<T>. Initially I have not done it because I allowed any method with the right signature. It is not required to implement a partikular iface. I have some examples, I'll centralize them in the doc.